src/native/libs/System.Native/pal_uid.h

```diff
@@ -82,3 +82,17 @@ PALEXPORT int32_t SystemNative_GetGroupList(const char* name, uint32_t group, ui
 * If the buffer is too small, errno is EINVAL.
 */
 PALEXPORT int32_t SystemNative_GetGroups(int32_t ngroups, uint32_t* groups);
+
+/**
+* Gets the user name associated with the specified group ID and stores it in the buffer.
+* On failure, returns a null char pointer and sets errno.
+* On success, returns a valid char pointer containing the group name.
+*/
+PALEXPORT char* SystemNative_GetGroupName(uint32_t gid);
+
+/**
+* Gets the user name associated with the specified user ID and stores it in the buffer.
+* On failure, returns a null char pointer and sets errno.
+* On success, returns a valid char pointer containing the user name.
+*/
+PALEXPORT char* SystemNative_GetUserName(uint32_t uid);
```

Comment:
cc @elinor-fung I'm updating the comments based on your suggestion:

```suggestion

/**
* Gets the user name associated with the specified group ID and stores it in the buffer.
* On failure, returns a null char pointer and sets errno.
* On success, returns a valid char pointer containing the group name.
* Note that this method returns new memory. Consumers can rely on the marshalling behaviour to free the returned string.
*/
PALEXPORT char* SystemNative_GetGroupName(uint32_t gid);

/**
* Gets the user name associated with the specified user ID and stores it in the buffer.
* On failure, returns a null char pointer and sets errno.
* On success, returns a valid char pointer containing the user name.
* Note that this method returns new memory. Consumers can rely on the marshalling behaviour to free the returned string.
*/
PALEXPORT char* SystemNative_GetUserName(uint32_t uid);
```

Reply:
I wouldn't mention anything about marshalling behavior here. The native API does something and how it is consumed does something too, but they are unrelated. Knowing how this API operates simply informs the consumer.